DSESwift/DataSyncer.swift

```diff
@@ -8,37 +8,398 @@
+                try DatabaseHandler.insertSensor(sensor)
+            } else {
+                try DatabaseHandler.updateSensor(sensor)
+            }
+        }
+    }
+    
+    private func getSensorsFromRemote() throws -> Array<Sensor>{
+        let downloadedArray = try SensorDataProxy.getSensors()
+        return convertAnyObjArrayToSensorArray(downloadedArray.arrayObject)
+    }
+    
+    private func convertAnyObjArrayToSensorArray(inputArray: Array<AnyObject>?) -> Array<Sensor>{
+        var sensors = Array<Sensor>()
+        for anyObj in inputArray! {
+            sensors.append(convertAnyObjectToSensor(anyObj))
+        }
+        return sensors
+    }
+    
+    private func convertAnyObjectToSensor(anyObj: AnyObject) -> Sensor{
+        let sensorDict = anyObj as! Dictionary<String, AnyObject>
+        let sourceName = sensorDict["source_name"] as! String
+        let sensorName = sensorDict["sensor_name"] as! String
+        let metaDict = sensorDict["meta"] as! Dictionary<String, AnyObject>
+        var sensorConfig = SensorConfig()
+        sensorConfig.meta = metaDict
         
-//closure 语法
-//        (parameters) -> returnType in
-//        
-//        statements
+        return Sensor(name: sensorName, source: sourceName, sensorConfig: sensorConfig, userId: KeychainWrapper.stringForKey(KEYCHAIN_USERID)!, remoteDataPointsDownloaded: false)
+    }
+    
+    func insertSensorDataIntoLocalDB(sensorData: JSON, sensorId: Int) throws {
+        for (_ ,subJson):(String, JSON) in sensorData["data"] {
+            //Do something you want
+            let value = JSONUtils.stringify(subJson["value"])
+            let time = NSDate(timeIntervalSince1970: subJson["time"].doubleValue / 1000.0)
+            let dataPoint = DataPoint(sensorId: sensorId, value: value, time: time)
+            try DatabaseHandler.insertOrUpdateDataPoint(dataPoint)
+        }
+    }
+    
+    func deleteDataIfExistsInRemoteAndExpired(id:Int) throws {
+        let persistentBoundary = NSDate().dateByAddingTimeInterval (-1 * persistentPeriod)
+        var queryOptions = QueryOptions()
+        queryOptions.endTime = persistentBoundary
+        queryOptions.existsInRemote = true
+        try DatabaseHandler.deleteDataPoints(id, queryOptions)
     }
-    func deletionInRemote() {
+    
+    func deleteDataIfExistsInRemote(id:Int) throws {
+        var queryOptions = QueryOptions()
+        queryOptions.existsInRemote = true
+        try DatabaseHandler.deleteDataPoints(id, queryOptions)
+    }
+    
+    func deleteDataIfExpired(id:Int) throws {
+        let persistentBoundary = NSDate().dateByAddingTimeInterval (-1 * persistentPeriod)
+        var queryOptions = QueryOptions()
+        queryOptions.endTime = persistentBoundary
+        try DatabaseHandler.deleteDataPoints(id, queryOptions)
+    }
+    
+    static func getJSONArray(dataPoints: Array<DataPoint>, sensorName: String) throws -> JSON{
```

Comment:
This should be in an util file, it is used as a static method but has nothing to do with the datasyncer